### ISSUE TITLE
Follow-Up for #411:  Should unavailable `Catalog`s be ignored during resolution?

### ISSUE THREAD

**Issue by everettraven** [operator-framework/operator-controller#428]
Mentioned in https://github.com/operator-framework/operator-controller/pull/411#discussion_r1332190866 by @joelanford:
>
> Follow-up: we probably should not ignore unavailable catalogs. If we do, we introduce non-determinism to resolution.
>
> If a Catalog that is needed for resolution is unavailable (that's all Catalogs right now because there's no way to disable catalogs globally or for a particular Operator), I think we should fail resolution, report the error, and then requeue when the Catalog changes state.

**Comment by joelanford** [comment]
Yeah, we need to think through implications. We likely need some way for an `Operator` to specify which catalogs should be used to resolve it.
